src/ooxml/java/org/apache/poi/xwpf/usermodel/XWPFChart.java

```diff
@@ -1,123 +1,343 @@
+    // lazy initialization
+    private Long checksum;
+    /**
+	 * this object is used to write embedded part of chart i.e. xlsx file in docx
+	 */
+    private OutputStream sheet;
+    /**
+     * this object is used to modify drawing properties
+     */
+	private CTInline ctInline;
+	
+	/**
+	 * constructor to
+	 * Create a new chart in document
+	 * 
+	 * @since POI 4.0
+	 */
+	protected XWPFChart() {
+		super();
+	}
+    /**
+     * Construct a chart from a package part.
+     *
+     * @param part the package part holding the chart data,
+     * the content type must be <code>application/vnd.openxmlformats-officedocument.drawingml.chart+xml</code>
+     *
+     * @since POI 4.0
+     */
+    protected XWPFChart(PackagePart part) throws IOException, XmlException {
+        super(part);
+    }
+
+    @Override
+    protected void commit() throws IOException {
+        XmlOptions xmlOptions = new XmlOptions(DEFAULT_XML_OPTIONS);
+        xmlOptions.setSaveSyntheticDocumentElement(new QName(CTChartSpace.type.getName().getNamespaceURI(), "chartSpace", "c"));
+
+        try (OutputStream out = getPackagePart().getOutputStream()) {
+            chartSpace.save(out, xmlOptions);
+        }
+    }
+
+    public Long getChecksum() {
+        if (this.checksum == null) {
+            InputStream is = null;
+            byte[] data;
+            try {
+                is = getPackagePart().getInputStream();
+                data = IOUtils.toByteArray(is);
+            } catch (IOException e) {
+                throw new POIXMLException(e);
+            } finally {
+                try {
+                    if (is != null) {
+                        is.close();
+                    }
+                } catch (IOException e) {
+                    throw new POIXMLException(e);
+                }
+            }
+            this.checksum = IOUtils.calculateChecksum(data);
+        }
+        return this.checksum;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        /**
+         * In case two objects ARE equal, but its not the same instance, this
+         * implementation will always run through the whole
+         * byte-array-comparison before returning true. If this will turn into a
+         * performance issue, two possible approaches are available:<br>
+         * a) Use the checksum only and take the risk that two images might have
+         * the same CRC32 sum, although they are not the same.<br>
+         * b) Use a second (or third) checksum algorithm to minimise the chance
+         * that two images have the same checksums but are not equal (e.g.
+         * CRC32, MD5 and SHA-1 checksums, additionally compare the
+         * data-byte-array lengths).
+         */
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+
+        if (!(obj instanceof XWPFChart)) {
+            return false;
+        }
+        return false;

```

Comment:
This is equivalent to the default equality by reference. Isn't it? 

But it is not compatible with the `hashCode` equality...